.github/workflows/build.yml

```diff
@@ -1,22 +1,42 @@
 # -----------------------------------------------------------------------------
 #   - invoked on push, pull_request, or manual trigger
-#   - test under at least 4 versions of python
+#   - test under all supported versions of Python
 # -----------------------------------------------------------------------------
 name: build
 on: [push, pull_request, workflow_dispatch]
 
 jobs:
+  get-python-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11  # Just used for running the versions script
+      - name: install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install packaging
+      - name: get python versions
+        id: set-matrix
+        run: |
+          echo "MATRIX_RESULT=$(python update_versions.py)" >> $GITHUB_ENV
+    outputs:
+      matrix: ${{ env.MATRIX_RESULT }}
   build:
+    needs: get-python-versions
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ${{ fromJSON(needs.get-python-versions.outputs.matrix) }}
     defaults:
       run:
         shell: bash -l {0}
     steps:
       - uses: actions/checkout@v2
-      - name: Set up Python ${{ "{{" }} matrix.python-version {{ "}}" }}
```

Comment:
This line is broken and was actually causing these builds to always break. I guess CVD never had builds passing.